### PR TITLE
fix(FOROME-634): all not mods for dtree func filters

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-634

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-inheritance-mode/modal-inheritance-mode.store.ts [hunk 1]
```diff
@@ -107,6 +117,27 @@ class ModalInheritanceModeStore {
     currentGroup
       .find((elem: any) => Array.isArray(elem))
       .map((item: string) => dtreeStore.addSelectedFilter(item))
+
+    this.isAllMode = currentGroup[2] === 'AND'
+    this.isNotMode = currentGroup[2] === 'NOT'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-inheritance-mode/modal-inheritance-mode.store.ts]: === ConditionJoinMode.**

File: src/pages/filter/ui/query-builder/ui/next-step-content-item.tsx [hunk 2]
```diff
@@ -91,7 +98,8 @@ export const NextStepContentItem = observer(
     const currentGroup = currentStep.groups[currNo]
     const isNumeric = currentGroup[0] === 'numeric'
 
-    const isNegateAttribute = currentGroup[2] === 'NOT'
+    const isNotMode = currentGroup[2] === 'NOT'
+    const isAllMode = currentGroup[2] === 'AND'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/ui/next-step-content-item.tsx:122]: same

File: src/utils/editStepAttribute.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const editStepAttribute = (
   const filteredAttribute = attribute.filter(
     (element: any) =>
       element !== 'OR' &&
-      element !== 'and' &&
+      element !== 'AND' &&
       element !== 'NOT' &&
```
Comment (AndreyUstyumenko) [inline src/utils/editStepAttribute.ts]: same

File: src/utils/editStepAttribute.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ export const editStepAttribute = (
...
       element !== 'or',
```
Comment (AndreyUstyumenko) [inline src/utils/editStepAttribute.ts:26]: is the lowercase correct?

File: src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts [hunk 5]
```diff
@@ -96,19 +101,28 @@ class ModalCompoundHetStore {
...
-  public fetchStatFunc(): void {
+  public fetchStatFunc(currentGroup: TFuncCondition): void {
     const params = `{"approx":${this.getApprox(this.approxCondition)},"state":${
       this.stateCondition === '-current-' || !this.stateCondition
         ? null
         : `"${this.stateCondition}"`
     }}`
 
+    if (currentGroup) {
+      this.isNotMode = currentGroup[2] === 'NOT'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-het/modal-compound-het.store.ts]: ConditionJoinMode.NOT

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts [hunk 6]
```diff
@@ -339,8 +341,14 @@ class ModalCompoundRequestStore {
         : `"${this.stateCondition}"`
     },"request":${requestString}}`
 
+    this.isNotMode = currentGroup[2] === 'NOT'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.store.ts]: same

File: src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.tsx [hunk 7]
```diff
@@ -77,7 +77,11 @@ export const ModalCompoundRequest = observer((): ReactElement => {
           handleSetCondition={handleSetCondition}
         />
 
-        <AllNotModalMods />
+        <AllNotModalMods
+          isNotModeChecked={modalCompoundRequestStore.isNotMode}
+          isNotModeDisabled={variants ? variants.length === 0 : true}
+          toggleNotMode={() => modalCompoundRequestStore.toggleMode()}
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-compound-request/modal-compound-request.tsx]: just {modalCompoundRequestStore.toggleMode}

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 8]
```diff
@@ -219,8 +222,14 @@ class ModalCustomInheritanceModeStore {
 
     const params = `{"scenario":${scenarioString}}`
 
+    this.isNotMode = currentGroup[2] === 'NOT'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: same

File: src/pages/filter/ui/modal-edit/components/modal-gene-region/modal-gene-region.store.ts [hunk 9]
```diff
@@ -39,6 +40,8 @@ class ModalGeneRegionStore {
 
     this.setLocusCondition(defaultValue)
 
+    this.isNotMode = currentGroup[2] === 'NOT'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-gene-region/modal-gene-region.store.ts]: same

File: src/utils/editStepAttribute.ts (hunk 4 above)
Comment (Lapk1n) [reply src/utils/editStepAttribute.ts:26]: Yup, there was some rare case that forced us to add this condition.

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by AlMaQntr — APPROVED** [review]

**Pull request merged**